### ISSUE TITLE
Compilation fails if std::tuple member in struct

### ISSUE THREAD

**Issue by SteveS-Serrano** [jorgen/json_struct#48]
I'm running on Linux Ubuntu 22.04. I ported the ```TEST_CASE("test_serialize_simple", "[json_struct][serialize]")``` in ```tests/json-struct-serialize-test.cpp``` to google test, and added members to the Simple struct. If I include the std::tuple member in the JS_OBJ list, the compiler generates reams of failure messages.

```
struct Simple
{
    std::string A;
    bool b;
    int some_longer_name;
    std::vector<int> vector;
    std::string string;
    std::tuple<std::string, float, int> tup;
    std::unordered_map<std::string, double> uo_map;
    std::optional<std::string> optional;

    //JS_OBJECT(JS_MEMBER(A), JS_MEMBER(b), JS_MEMBER(some_longer_name));
    //JS_OBJ(A, b, some_longer_name, vector, string, tup, uo_map, optional); /// <<<< this fails
    JS_OBJ(A, b, some_longer_name, vector, string, uo_map, optional);        /// <<<< this succeeds

};

TEST_F(EsiJsonSerdesGTest, GenerateSimple)
{
    Simple simple;
    simple.A = "TestString";
    simple.b = false;
    simple.some_longer_name = 456;
    simple.vector = {1,2,3,4,5,6,7,8,9};
    simple.unordered_map = { {"foo", 100.0}, {"bar", 101.1}};
    simple.optional = "hello";

    std::string output = JS::serializeStruct(simple, JS::SerializerOptions(JS::SerializerOptions::Compact));
    EXPECT_STREQ(output.c_str(), expected1_compact);
}
```
* git commit hash
```
$ git log -n 1
commit 9b719771470536763430703b7f04acc558dafe56 (HEAD -> master, origin/master, origin/HEAD)
Author: Jørgen Lind <jorgen.lind@gmail.com>
Date:   Thu Jan 19 12:12:19 2023 +0100

    Fix some warnings
$
```

* OS
```
$ uname -a
Linux demo 5.15.0-70-generic #77-Ubuntu SMP Tue Mar 21 14:02:37 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.1 LTS
Release:        22.04
Codename:       jammy
$
```
* compiler
```
$ g++ --version
g++ (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0
Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

**Comment by SteveS-Serrano** [comment]
I also ported the file ```tests/json-struct-serialize-tuple.cpp``` to our google test environment. This test compiles and runs ok. 

However, I noticed that it only serializes tuples of type ```JS::Tuple```. The first test, ```serialize_tuple``` serializes a struct with a ```JS::Tuple``` member. The second test, ```bool_tuple```, which uses ```std::tuple```, only parses json into a struct, it doesn't serialize the struct containing the ```std::tuple``` member.

Adding this line to the end of the ```bool_tuple``` test causes compilation to fail:

```
  std::string bar = JS::serializeStruct(tbStruct);
```

The complete Google Test version of the test is:

```
struct TestInt {
  std::tuple<int32_t> member;
  JS_OBJ(member);
};

struct TestBool {
  std::tuple<bool> member;
  JS_OBJ(member);
};

TEST_F(EsiJsonStructTest, bool_tuple)
{
  TestInt tiStruct;
  JS::ParseContext intContext(R"({ "member": [5] })");
  EXPECT_EQ(intContext.parseTo(tiStruct), JS::Error::NoError);

  TestBool tbStruct;
  JS::ParseContext boolContext(R"({ "member": [true] })");
  EXPECT_EQ(boolContext.parseTo(tbStruct), JS::Error::NoError);

  std::string bar = JS::serializeStruct(tbStruct);    // <<<< compilation fails

}
```

**Comment by jorgen** [comment]
Ok, I have fixed this now with: e3045af90ff331c97cc8f71374f558ae13409d9f. I have a copy/paste bug in the serialisation code for std::tuple. Closing the issue. Please feel free to reopen if your feel the issue isn't fixed.
